### PR TITLE
ECC Enc test: fix ecc_ctx_kdf_salt_test

### PR DESCRIPTION
**Pull request by SparkiDev**
# Description

Test had uninitialized memory read as the plaintext buffer wasn't memset to AES block size.
Test leaked when HAVE_WOLF_BIGINT is defined as keys weren't freed before initialization.
Test function only compiled when AES-CTR compiled in when it uses AES-CBC.
Made function return a unique value.
Other tidying up.

# Testing

./configure '--disable-shared' 'CFLAGS=-DHAVE_WOLF_BIGINT' '--enable-eccencrypt' '--enable-debug' '--disable-dh' '--disable-rsa'
valgrind ./wolfcrypt/test/testwolfcrypt

Need --enable-aesctr to make unmodified code compile the function: ecc_ctx_kdf_salt_test()

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfcrypt/test/test.c [hunk 1]
```diff
@@ -26606,80 +26606,90 @@ static int ecc_ctx_kdf_salt_test(WC_RNG* rng, ecc_key* a, ecc_key* b)
...
+    ret = wc_ecc_init(a);
+    if (ret != 0)
+        ret = -10480;
+    if (ret == 0) {
+        ret = wc_ecc_init(b);
+        if (ret != 0)
+            ret = -10481;
+    }
 
+    ret = wc_ecc_make_key(rng, 32, a);
```
Comment (douzzer) [inline wolfcrypt/test/test.c:26633]: why isn't this conditional on `(ret == 0)`?

File: wolfcrypt/test/test.c [hunk 2]
```diff
@@ -26606,80 +26606,90 @@ static int ecc_ctx_kdf_salt_test(WC_RNG* rng, ecc_key* a, ecc_key* b)
...
+    }
 
     XMEMCPY(plaintext, message, XSTRLEN(message));
-
-    while (plaintextLen % AES_BLOCK_SIZE != 0) {
-        plaintextLen++;
-    }
+    plaintextLen = (((word32)XSTRLEN(message) + AES_BLOCK_SIZE - 1) /
+        AES_BLOCK_SIZE) * AES_BLOCK_SIZE;
+    XMEMSET(plaintext + XSTRLEN(message), 0, plaintextLen - XSTRLEN(message));
```
Comment (douzzer) [inline wolfcrypt/test/test.c:26666]: sanitizer is not happy with this construction.  probably just do what I did in #6081 -- pre-zero `plaintext` in its entirety, `XMEMSET(plaintext, 0, 128);`  I retested with that change and it was fine.
```
In file included from /usr/include/string.h:535,
                 from ./wolfssl/wolfcrypt/types.h:623,
                 from wolfcrypt/test/test.c:34:
In function ‘memset’,
    inlined from ‘ecc_ctx_kdf_salt_test’ at wolfcrypt/test/test.c:26666:5,
338d8db274 (<106998124+jpbland1@users.noreply.github.com> 2023-02-10 13:05:52 -0500 27333)         ret = ecc_ctx_kdf_salt_test(&rng, userA, userB);
    inlined from ‘ecc_encrypt_test’ at wolfcrypt/test/test.c:27333:15:
/usr/include/bits/string_fortified.h:59:10: error: ‘__builtin_memset’ offset [0, 1] is out of the bounds [0, 0] [-Werror=array-bounds]
   59 |   return __builtin___memset_chk (__dest, __ch, __len,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   60 |                                  __glibc_objsize0 (__dest));
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/bits/string_fortified.h:59:10: error: ‘__builtin_memset’ offset [0, 1] is out of the bounds [0, 0] [-Werror=array-bounds]
```

found by the `sanitizer-all-asm-smallstack-async` test, i.e. `'--enable-all' '--enable-asynccrypt' '--enable-intelasm' '--enable-sp-math-all' '--enable-smallstack' '--enable-smallstackcache' 'CPPFLAGS=-pedantic' 'CC=gcc' 'CFLAGS=-g -fno-omit-frame-pointer -fsanitize=address,pointer-subtract,leak,undefined,float-cast-overflow,float-divide-by-zero,bounds-strict -fsanitize-recover=all --param=max-vartrack-size=128000000 -march=native' 'LDFLAGS=-g -fno-omit-frame-pointer -fsanitize-recover=all -fsanitize=address,pointer-subtract,leak,undefined,float-cast-overflow,float-divide-by-zero,bounds-strict -fno-sanitize-recover=all '`

the async part has nothing to do with it AFAIK -- all about the smallstack.

**Review by douzzer — CHANGES_REQUESTED** [review]
can you merge into this PR the fixes to return codes in #6081 ?

File: wolfcrypt/test/test.c (hunk 2 above)
Comment (dgarske) [reply wolfcrypt/test/test.c:26666]: See: https://github.com/wolfSSL/wolfssl/pull/6081/files#diff-28520d015f1d43f2061e6286aea6efc713b4dba91818eef1206611acb817bd13R26681

**Comment by douzzer** [comment]
#6081 is merged now, with the additional fixes from #6083.

I ran `wolfssl-multi-test.sh ... fips-140-2-optest fips-140-3-optest-acvp-sp-asm linuxkm-all-fips-140-3-dev-dyn-hash intmath-sanitizer sanitizer-all-asm-smallstack-async sanitizer-all-asm-smallstack-async-quic` on `master` at 6877c98d82 afterwards, in an abundance of caution, and it was all clean.

**Pull request closed without merging**